File: src/test/java/kitchenpos/fixture/application/OrderLineItemFixture.java
package kitchenpos.fixture.application;

import kitchenpos.domain.Menu;
import kitchenpos.domain.OrderLineItem;

import java.math.BigDecimal;
import java.util.UUID;

public class OrderLineItemFixture {

    public static OrderLineItem createOrderLineItem(UUID menuId, int price, long quantity) {
        OrderLineItem orderLineItem = new OrderLineItem();
        orderLineItem.setMenuId(menuId);
        orderLineItem.setPrice(BigDecimal.valueOf(price));
        orderLineItem.setQuantity(quantity);
        return orderLineItem;
    }

    public static OrderLineItem createOrderLineItem(long quantity) {
        OrderLineItem orderLineItem = new OrderLineItem();
        orderLineItem.setQuantity(quantity);
        return orderLineItem;
    }

    public static OrderLineItem createOrderLineItem(int price) {
        OrderLineItem orderLineItem = new OrderLineItem();
        orderLineItem.setPrice(BigDecimal.valueOf(price));
        return orderLineItem;
    }

    public static OrderLineItem createOrderLineItem(UUID menuId, Menu menu) {
        OrderLineItem orderLineItem = new OrderLineItem();
        orderLineItem.setMenuId(menuId);
        orderLineItem.setMenu(menu);
        orderLineItem.setPrice(BigDecimal.valueOf(20000));
        orderLineItem.setQuantity(1);
        return orderLineItem;
    }
}

File: src/test/kotlin/kitchenpos/order/OrderLineItemFixture.kt
package kitchenpos.order

import kitchenpos.domain.Menu
import kitchenpos.domain.Order
import kitchenpos.domain.OrderLineItem
import java.math.BigDecimal

class OrderLineItemFixture {
    companion object {
        fun fixture(
            order: Order,
            menu: Menu,
            quantity: Long,
        ): OrderLineItem {
            val orderLineItem = OrderLineItem()
            orderLineItem.seq = order.orderLineItems.size.toLong()
            orderLineItem.menu = menu
            orderLineItem.quantity = quantity
            orderLineItem.menuId = menu.id
            orderLineItem.price = menu.price.multiply(BigDecimal.valueOf(quantity))
            return orderLineItem
        }
    }
}
